zimcheck complains about geo: links, it should not

on bm.wikipedia.org

```
[ERROR] Invalid internal links found :
  A/geo:-36.772777777777776,-73.06305555555555 (geo:-36.772777777777776,-73.06305555555555) was not found in article A/Concepción
  A/geo:51.40833333333333,22.055555555555557 (geo:51.40833333333333,22.055555555555557) was not found in article A/Końskowola
  A/geo:-25.429722222222225,-49.27194444444444 (geo:-25.429722222222225,-49.27194444444444) was not found in article A/Kuritiba
```

But in the HTML:
```
<a href="geo:-25.429722222222225,-49.27194444444444" class="external text"
```

So, the HTML looks good, zimcheck should not complain IMO.

See also https://github.com/openzim/mwoffliner/issues/771

--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,13 @@
+zim-tools 1.0.3
+===============
+
+ * ZimCheck handles properly geo: mailto: tel: javascript: links
+
 zim-tools 1.0.2
 ===============
 
  * Update readme.
- * ZimCheck return a proper status code.
+ * ZimCheck returns a proper status code.
  * Update to last API of libzim 5.0.0
  * Temporally deactivate compilation of zimBench on windows.
 

--- a/src/zimcheck.cpp
+++ b/src/zimcheck.cpp
@@ -162,12 +162,15 @@ int adler32(std::string buf)                        //Adler32 Hash Function. Use
 
 inline bool isExternalUrl(const std::string& input_string)
 {
-    // A string starting with "<scheme>://".
-    static std::regex external_url_regex = std::regex("[^:/?#]+:\\/\\/.*", std::regex_constants::icase);
+    // A string starting with "<scheme>://" or "geo:" or "tel:" or "javascript:" or "mailto:"
+    static std::regex external_url_regex =
+      std::regex("([^:/?#]+:\\/\\/|geo:|tel:|mailto:|javascript:).*",
+                 std::regex_constants::icase);
     return std::regex_match(input_string, external_url_regex);
 }
 
-inline bool isInternalUrl(const std::string& input_string)                 //Checks if a URL is an internal URL or not. Uses RegExp.
+// Checks if a URL is an internal URL or not. Uses RegExp.
+inline bool isInternalUrl(const std::string& input_string)
 {
     return !isExternalUrl(input_string);
 }

